src/transformers/modeling_flax_utils.py

```diff
@@ -129,7 +142,20 @@ def params(self, params: Union[Dict, FrozenDict]):
                 "Some parameters are missing. Make sure that `params` include the following "
                 f"parameters {self.required_params - param_keys}"
             )
-        self._params = freeze(params)
+        self._params = params
+
+    def tie_weights(self):
+        output_embeddings = self.get_output_embeddings()
+
+        # tie word embedding with output embeddings
+        if output_embeddings is not None and self.config.tie_word_embeddings:
+            input_embbeddings_key = self.get_input_embeddings_key()
+            flattened_dict = flatten_dict(self.params)
+
+            # transpose weight and set correctly
+            flattened_dict[input_embbeddings_key] = jnp.asarray(output_embeddings.T)
```

Comment:
Sorry for barging in, but I think this is correct, this will just initialized as tied but since jax arrays are immutable new copies will be created on gradient update that are not longer tied. The way they do it in the flax examples is to share a layer (see [this](https://github.com/google/flax/blob/a6282dba2f2be8d1724ea35f6e1d963097058cd9/examples/wmt/models.py#L342)), not sure how it would best integrate with the naming scheme used for pytorch

Reply:
Yeah great catch! I realized this yesterday as well & [here](https://github.com/google/flax/discussions/1264#discussioncomment-644887) is a discussion here on the Flax repo regarding this point. The solution above is temporary so that at least during inference the model behaves correctly, *e.g.*, without this "hack" the following could be incorrect during inference:

```python 
bert = FlaxBertForMaskedLM.from_pretrained("some bert model that is tied", from_pt=True)
```

=> I'll add a big message stating the problem